src/com/google/javascript/jscomp/PeepholeFoldConstants.java

```diff
@@ -1080,6 +1081,15 @@ static TernaryValue evaluateComparison(int op, Node left, Node right, boolean us
       case Token.NUMBER:
         if (right.isNumber()) {
           return compareAsNumbers(op, left, right, useTypes);
+        } else if (right.isString()) {
+          return compareAsNumbers(op, left, IR.number(NodeUtil.getNumberValue(right)), useTypes);
```

Comment:
I added tests for these and made the code explicit. Added a couple of links to the ES6 specs.